iam/snippets/snippets.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	iam "google.golang.org/api/iam/v1"
+)
+
+var client, _ = google.DefaultClient(
+	context.Background(),
+	iam.CloudPlatformScope)
+var iamService, _ = iam.New(client)
+
+// [START iam_view_grantable_roles]
+func viewGrantableRoles(fullResourceName string) {
+	request := iam.QueryGrantableRolesRequest{FullResourceName: fullResourceName}
+	response, _ := iamService.Roles.QueryGrantableRoles(&request).Do()
```

Comment:
Each snippet should create the service.

Reply:
This makes the snippets long and repetitive, especially in docs with lots of snippets demonstrating very similar tasks. And as I said earlier, it deviates from the approach used by every other language.